pace-util/pace/util/checkpointer/snapshots.py

```diff
@@ -0,0 +1,72 @@
+import collections
+
+import numpy as np
+
+from pace.util._optional_imports import xarray as xr
+
+from .base import Checkpointer
+
+
+def make_dims(savepoint_dim, label, data_list):
+    """
+    Helper which defines dimension names for an xarray variable.
+
+    Used to ensure no dimensions have the same name but different sizes
+    when defining xarray datasets.
+    """
+    data = np.concatenate([array[None, :] for array in data_list], axis=0)
+    dims = [savepoint_dim] + [f"{label}_dim{i}" for i in range(len(data.shape[1:]))]
+    return dims, data
+
+
+class Snapshots:
+    def __init__(self):
+        self._savepoints = collections.defaultdict(list)
+        self._python_data = collections.defaultdict(list)
+
+    def compare(self, savepoint_name: str, variable_name: str, python_data):
```

Comment:
`compare` doesn't quite seem like the right name for this method to me. We're adding data to the snapshot, not looking at the differences between two datasets inside this method. `add_data` isn't as eloquent but something like that might be clearer

Reply:
I'll update it. This is a relic of when it was actually aligning Python and Fortran data (similarly to the _python_data name) and I wasn't careful because it's private API.

Reply:
Cool, thanks.